twoinputs/demand_uncertainty.ipynb

```diff
@@ -0,0 +1,314 @@
+    "dataframe = pd.read_csv(\"../data/uiuc_demand_data.csv\", usecols=['time','kw'])\n",
+    "demand = np.array(dataframe.kw).astype('float64')\n",
+    "\n",
+    "# normalize\n",
+    "norm = np.linalg.norm(demand)\n",
+    "demand_norm = demand/norm"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Norm of elevation angles: 7170.040655802196\n",
+      "Norm of demand: 10142199.043147163\n"
+     ]
+    }
+   ],
+   "source": [
+    "time_range = pd.date_range(start='1/1/2015', end='7/1/2019', freq='h')[:-2]\n",
+    "elevation = generate_elevation_series(time_range, timestamps=True)\n",
+    "norm_a = np.linalg.norm(elevation)\n",
+    "angles_norm = elevation/norm_a\n",
+    "print(f\"Norm of elevation angles: {norm_a}\")\n",
+    "print(f\"Norm of demand: {norm}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Shape of X_in is (2, 39407)\n"
+     ]
+    }
+   ],
+   "source": [
+    "X_in = np.concatenate([[demand_norm, angles_norm]], axis=1)\n",
+    "    \n",
+    "print(f\"Shape of X_in is {X_in.shape}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# optimal parameters from demand-windowsize\n",
+    "params = {'n_reservoir':1000, \n",
+    "          'sparsity':0.03, \n",
+    "          'rand_seed':85, \n",
+    "          'rho':1.5, \n",
+    "          'noise':0.0001, \n",
+    "          'future':72, \n",
+    "          'window':24, \n",
+    "          'trainlen':22400}\n",
+    "\n",
+    "rand_seeds = [85, 21, 22, 30, 122, 5, 79]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "futureTotal = params['future']\n",
+    "time_range = np.array(dataframe.time)\n",
+    "hour_range = np.arange(0, len(time_range), 1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Running Simulation 0\n",
+      "Running Simulation 1\n",
+      "Running Simulation 2\n",
+      "Running Simulation 3\n",
+      "Running Simulation 4\n",
+      "Running Simulation 5\n",
+      "Running Simulation 6\n",
+      "This simulation took 84.91 seconds\n",
+      "This simulation took 1.42 minutes\n"
+     ]
+    }
+   ],
+   "source": [
+    "N = len(rand_seeds)\n",
+    "predictions = np.ones((N, futureTotal))\n",
+    "\n",
+    "tic = time.perf_counter()\n",
+    "\n",
+    "for x, seed in enumerate(rand_seeds):\n",
+    "    print(f\"Running Simulation {x}\")\n",
+    "    params['rand_seed'] = seed\n",
+    "    prediction = esn_prediction(X_in.T, params)\n",
```

Comment:
Consider changing this variable name; while scanning over I got it mixed it up with predictions.